session.go

```diff
@@ -836,10 +836,24 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
 		return
 	}
 
-	s.PrepareTxnCtx(goctx.Background())
+	goCtx := goctx.Background()
+	inTxn := s.GetSessionVars().InTxn()
+	isAutocommit := s.GetSessionVars().IsAutocommit()
```

Comment:
I think we don't need to consider auto-commit.
We only care about  `inTxn`.
prepare should not start a new transaction.

Reply:
You are right. Prepare statement should not start a new transaction. Fixed PTAL